pkg/service/collector.go

```diff
@@ -62,9 +79,15 @@ func NewCollectorService(jaeger *v1.Jaeger, selector map[string]string) *corev1.
 			},
 		},
 	}
+
 }
 
 // GetNameForCollectorService returns the service name for the collector in this Jaeger instance
 func GetNameForCollectorService(jaeger *v1.Jaeger) string {
 	return fmt.Sprintf("%s-collector", jaeger.Name)
 }
+
+// GetNameForHeadlessCollectorService returns the headless service name for the collector in this Jaeger instance
+func GetNameForHeadlessCollectorService(jaeger *v1.Jaeger) string {
+	return fmt.Sprintf("%s-collector-headless", jaeger.Name)
```

Comment:
nit: This could take the output from `GetNameForCollectorService` and append `-headless`. Only minor though feel free to ignore.